public/app/plugins/datasource/cloud-monitoring/components/MetricQueryEditor.tsx

```diff
@@ -64,70 +50,65 @@ function Editor({
   onRunQuery,
   customMetaData,
   variableOptionGroup,
+  editorMode,
 }: React.PropsWithChildren<Props>) {
-  const [state, setState] = useState<State>(defaultState);
-  const { projectName, metricType, groupBys, editorMode, crossSeriesReducer } = query;
-
-  useEffect(() => {
-    if (projectName && metricType) {
-      datasource
-        .getLabels(metricType, refId, projectName)
-        .then((labels) => setState((prevState) => ({ ...prevState, labels })));
-    }
-  }, [datasource, groupBys, metricType, projectName, refId, crossSeriesReducer]);
-
-  const onChange = useCallback(
-    (metricQuery: MetricQuery | SLOQuery) => {
-      onQueryChange({ ...query, ...metricQuery });
+  const onChangeTimeSeriesList = useCallback(
+    (timeSeriesList: TimeSeriesList) => {
+      onQueryChange({ ...query, timeSeriesList });
       onRunQuery();
     },
     [onQueryChange, onRunQuery, query]
   );
 
-  const onMetricTypeChange = useCallback(
-    ({ valueType, metricKind, type }: MetricDescriptor) => {
-      const preprocessor =
-        metricKind === MetricKind.GAUGE || valueType === ValueTypes.DISTRIBUTION
-          ? PreprocessorType.None
-          : PreprocessorType.Rate;
-      const { perSeriesAligner } = getAlignmentPickerData(valueType, metricKind, state.perSeriesAligner, preprocessor);
-      onChange({
-        ...query,
-        perSeriesAligner,
-        metricType: type,
-        valueType,
-        metricKind,
-        preprocessor,
-      });
+  const onChangeTimeSeriesQuery = useCallback(
+    (timeSeriesQuery: TimeSeriesQuery) => {
+      onQueryChange({ ...query, timeSeriesQuery });
+      onRunQuery();
     },
-    [onChange, query, state]
+    [onQueryChange, onRunQuery, query]
   );
```

Comment:
all this code was moved to `VisualMetricQueryEditor` since it was only relevant there